05-onchain.md

```diff
@@ -185,231 +161,192 @@ Similarly, when node A sees a *commitment transaction* from B:
+There can be HTLCs that are not represented by an output: either
+because they were trimmed as dust or because the receipient has two
+valid commitment transactions, and the HTLCs differ in each.
 
 ## Requirements
 
-
 If the HTLC output is spent using the payment preimage, the HTLC
 output is considered *irrevocably resolved*, and the node MUST extract
 the payment preimage from the transaction input witness.
 
-
 If the HTLC output has *timed out* and not been *resolved*, the node
 MUST *resolve* the output and MUST fail the corresponding incoming
 HTLC (if any) once the resolving transaction has reached reasonable
-depth.  If the transaction is the node's own
+depth. If the transaction is the node's own
 commitment transaction, it MUST *resolve* the output by spending it
 using the HTLC-timeout transaction, and the HTLC-timeout
 transaction output MUST be *resolved* as described in "On-chain HTLC
-Transaction Handling".  Otherwise it MUST resolve the
+Transaction Handling". Otherwise it MUST resolve the
 output by spending it to a convenient address.
 
-
-For any committed HTLC which does not have an output in this
+For any committed HTLC that does not have an output in this
 commitment transaction, the node MUST fail the corresponding incoming
 HTLC (if any) once the commitment transaction has reached reasonable
 depth, and MAY fail it sooner if no valid commitment transaction
 contains an output corresponding to the HTLC.
 
-
 ## Rationale
 
-
-If the commitment transaction is theirs, the only way to spend the
+If the commitment transaction is the recipient's, the only way to spend the
 HTLC output using a payment preimage is for them to use the
-HTLC-success transaction.  If the commitment transaction is ours, they
+HTLC-success transaction. If the commitment transaction is the offerer's, the recipient
 could create any transaction using the preimage.
 
-
-The payment preimage either serves to prove payment (if this node
-originated the payment), or to redeem the corresponding incoming HTLC
-from another peer.  Note that we don't care about the fate of the
-HTLC-spending transaction itself once we've extracted the payment
+The payment preimage either serves to prove payment (for the offerer originating the payment) or
+to redeem the corresponding incoming HTLC from another peer (for an offerer forwarding the payment). Note that the parties don't care about the fate of the
+HTLC-spending transaction itself once they've extracted the payment
 preimage; the knowledge is not revocable.
 
-
-Note that in cases where both resolutions are possible (payment
-success seen after timeout, for example), either interpretation is
-acceptable; it is the responsibility of the other node spend it
+In cases where both resolutions are possible (when payment
+success is seen after timeout, for example), either interpretation is
+acceptable; it is the responsibility of the recipient to spend it
 before this occurs.
```

Comment:
...possible (e.g. when a node receives payment success after timeout)...

Reply:
It's unclear to me which node this refers to: sender, receiver, or both?

Reply:
Changed; flagging for @rustyrussell in case any additional clarification is necessary. (I think the "a node" phrase now used is correct.)